Assets/LeapMotionModules/Interaction/Examples/Buttons/InteractionSlider.cs

```diff
@@ -1,173 +1,62 @@
-    [Tooltip("The minimum and maximum heights the button can exist at.")]
-    public Vector2 MinMaxHeight = new Vector2(0f, 0.02f);
-    [Tooltip("The height that this button rests at; this value is a lerp in between the min and max height.")]
-    public float RestingHeight = 0.5f;
-    [Tooltip("The minimum and maximum horizontal extents that the slider can slide to.")]
-    public Vector2 HorizontalSlideLimits = new Vector2(0f, 0f);
-    [Tooltip("The minimum and maximum vertical extents that the slider can slide to.")]
-    public Vector2 VerticalSlideLimits = new Vector2(0f, 0f);
-
-    //State Events
-    public UnityEvent OnPress = new UnityEvent();
-    public UnityEvent OnUnpress = new UnityEvent();
+  /** A physics-enabled slider. Sliding is triggered by physically pushing the slider to its compressed position. 
+  *  Increasing the horizontal and vertical slide limits allows it to act as either a 1D or 2D slider
+  */
+  public class InteractionSlider : InteractionButton {
+    [Space][Space]
 
-    //Public State variables
-    public bool isDepressed { get; protected set; }
-    public bool depressedThisFrame { get; protected set; }
-    public bool unDepressedThisFrame { get; protected set; }
+    [Tooltip("The minimum and maximum values that the slider reports on the horizontal axis.")]
+    public Vector2 HorizontalValueRange = new Vector2(0f, 1f);
+    [Tooltip("The minimum and maximum values that the slider reports on the horizontal axis.")]
+    public Vector2 VerticalValueRange = new Vector2(0f, 1f);
 
-    //Private State Variables
-    private InteractionBehaviour behaviour;
-    private LeapGraphic element;
-    private Rigidbody body;
-    private Rigidbody lastDepressor;
-    private Vector3 localDepressor;
-    private Vector3 InitialLocalPosition;
-    private Vector3 PhysicsPosition = Vector3.zero;
-    private Vector3 PhysicsVelocity = Vector3.back;
-    private Vector3 localPhysicsPosition;
-    private bool physicsOccurred = false;
-    private bool handChirality = false;
+    [Space]
 
-    void Start() {
-      //Initialize Elements
-      behaviour = GetComponent<InteractionBehaviour>();
-      element = GetComponent<LeapGraphic>();
-      body = behaviour.rigidbody;
+    [Tooltip("The minimum and maximum horizontal extents that the slider can slide to in world space.")]
+    [MinMax(-0.25f, 0.25f)]
+    public Vector2 HorizontalSlideLimits = new Vector2(0f, 0f);
+    [MinMax(-0.25f, 0.25f)]
+    [Tooltip("The minimum and maximum vertical extents that the slider can slide to in world space.")]
+    public Vector2 VerticalSlideLimits = new Vector2(0f, 0f);
 
-      //Initialize Positions
-      InitialLocalPosition = transform.localPosition;
-      transform.localPosition = InitialLocalPosition;
-      localPhysicsPosition = InitialLocalPosition;
-      PhysicsPosition = transform.position;
-      body.position = PhysicsPosition;
+    [HideInInspector]
+    public float HorizontalSliderValue;
```

Comment:
This should probably be a property, since you probably don't want this to be serialized, and probably do want to do custom behavior when the user sets it's value (like moving the obj to the correct place)